nixos/modules/services/misc/irkerd.nix

```diff
@@ -0,0 +1,67 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.irkerd;
+  ports = [ 6659 ];
+in
+{
+  options.services.irkerd = {
+    enable = mkOption {
+      description = "Whether to enable irker service.";
+      default = false;
+      type = types.bool;
+    };
+
+    openPorts = mkOption {
+      description = "Open ports in the firewall for irkerd";
+      default = false;
+      type = types.bool;
+    };
+
+    listenAddress = mkOption {
+      default = "localhost";
+      example = "0.0.0.0";
+      type = types.str;
+      description = ''
+          Specifies the bind address on which the irker daemon listens.
+          The default is localhost.
+
+          Irker authors strongly warn about the risks of running this on
+          a publicly accessible interface, so change this with caution.
+      '';
+    };
+
+    nick = mkOption {
+      default = "irker";
+      type = types.str;
+      description = "Nick to use for irker";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.irkerd = {
+      description = "Internet Relay Chat (IRC) notification daemon";
+      requires = [ "network.target" ];
```

Comment:
I had the same thought (at least regarding using `after`), but went with `requires` since that's what their provided systemd service file does.  I'm not sure what the exact semantic differences are between the two, and since it seems `after` is the common situation I'd default to using that unless there's a good reason for the other?  Anyway I'll update accordingly shortly.